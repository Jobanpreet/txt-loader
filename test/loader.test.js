import { compiler, assetLoader, modulePrefixer } from './helpers';

describe('Txt file loader', () => {
  let fileData;
  describe('GIVEN esModule is false', () => {
    beforeEach(async () => {
      fileData = await assetLoader('./test/assets/whats-your-name.txt');
    });

    it('SHOULD return valid response', async () => {
      const esModule = false;
      const options = {
        entryPath: './test/index.js',
        esModule,
      };

      const output = await compiler(options);

      expect(output.toJson().modules[0].source).toBe(

        `${modulePrefixer(esModule)}"${fileData}"`,
      );
    });
  });

  describe('GIVEN esModule is true', () => {
    beforeEach(async () => {
      fileData = await assetLoader('./test/assets/whats-your-name.txt');
    });

    it('SHOULD return valid response', async () => {
      const esModule = true;
      const options = {
        entryPath: './test/index.js',
        esModule,
      };

      const output = await compiler(options);

      expect(output.toJson().modules[0].source).toBe(
        `${modulePrefixer(esModule)}"${fileData}"`,
      );
    });
  });

  describe('GIVEN options are invalid', () => {
    it('SHOULD throw error', async () => {
      let error;
      const options = {
        entryPath: './test/index.js',
        esModule: 'true',
      };
      try {
        const output = await compiler(options);
      } catch (e) {
        error = e;
      }

      expect(error).toMatchSnapshot();
    });
  });
});
