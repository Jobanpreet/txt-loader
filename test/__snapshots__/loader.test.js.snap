// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Txt file loader GIVEN options are invalid SHOULD throw error 1`] = `
[Error: ./test/assets/whats-your-name.txt
Module build failed (from ./src/index.js):
ValidationError:  Invalid Options

options.esModule should be boolean

    at validateOptions (/Users/Joban/Desktop/Joban/Projects/UI/webpack-demo/new-txt-loader/txt-loader/node_modules/schema-utils/src/validateOptions.js:32:11)
    at Object.txtLoader (/Users/Joban/Desktop/Joban/Projects/UI/webpack-demo/new-txt-loader/txt-loader/src/index.js:8:3)
 @ ./test/index.js 1:0-47 3:15-18]
`;
